apiclient/harvester_api/models.py

```diff
@@ -292,6 +292,27 @@ def add_network(self, name, net_uid, model="virtio", mac_addr=None):
         self.networks.append(net)
         return net
 
+    def replace_default_network(self, name, net_uid, model="virtio", mac_addr=None):
+        for k in range(len(self.networks)):
+            if self.networks[k]["network"]["name"] == "default":
+                self.networks.pop(k)
+
+        return self.add_network(name, net_uid, model, mac_addr)
+
+    def add_ssh_public_key(self, public_key):
+        userdata = yaml.safe_load(self.user_data) or dict()
+        userdata["ssh_authorized_keys"] = [public_key]
+
+        self.user_data = "#cloud-config\n" + yaml.dump(userdata)
+
+    def add_password(self, password):
```

Comment:
Spec class is reflect to UI's input, so we will not add handy method like this, user should assign `userdata` directly.